mirrord/kube/src/resolved.rs

```diff
@@ -10,25 +10,40 @@ use mirrord_config::{feature::network::incoming::ConcurrentSteal, target::Target
 use tracing::Level;
 
 use super::{
-    api::{
-        kubernetes::{get_k8s_resource_api, rollout::Rollout},
-        runtime::RuntimeData,
-    },
+    api::{kubernetes::get_k8s_resource_api, runtime::RuntimeData},
     error::KubeApiError,
 };
+use crate::api::{kubernetes::rollout::Rollout, runtime::RuntimeDataFromLabels};
+
+pub mod cron_job;
+pub mod deployment;
+pub mod job;
+pub mod pod;
+pub mod resource;
+pub mod rollout;
+pub mod stateful_set;
 
 /// Helper struct for resolving user-provided [`Target`] to Kubernetes resources.
 #[derive(Debug, Clone)]
 pub enum ResolvedTarget {
-    Deployment(Deployment, Option<String>),
-    Rollout(Rollout, Option<String>),
-    Job(Job, Option<String>),
-    CronJob(CronJob, Option<String>),
-    StatefulSet(StatefulSet, Option<String>),
-    Pod(Pod, Option<String>),
+    Deployment(ResolvedResource<Deployment>),
+    Rollout(ResolvedResource<Rollout>),
+    Job(ResolvedResource<Job>),
+    CronJob(ResolvedResource<CronJob>),
+    StatefulSet(ResolvedResource<StatefulSet>),
+    Pod(ResolvedResource<Pod>),
     Targetless(String),
 }
 
+#[derive(Debug, Clone)]
+pub struct ResolvedResource<R>
```

Comment:
Wondering if we even need `ResolvedTarget` if we have this struct.

Reply:
Maybe not, if we convert the `config::Target` to a typed `ResolvedResource<Deployment>`, for example, but this could get a bit messy to initialize.

- A simplified example of this type creation (converting directly from `Target`):

```rust
impl<R> ResolvedResource<R> {
  // Warning: returning different types!
  fn new(target: config::Target) -> Self {
    match target { 
      Target::Deployment(dep) => ResolvedResource { resource: dep },
      Target::Job(job) => ResolvedResource { resource: job },
    }
  }
}
```